File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--add-content-controls-to-word-documents_6.cs
        private System.Collections.Generic.List
            <Microsoft.Office.Tools.Word.RichTextContentControl> richTextControls;

        private void Application_DocumentOpen(Microsoft.Office.Interop.Word.Document Doc)
        {
            if (Doc.ContentControls.Count > 0)
            {
                Document extendedDocument = Globals.Factory.GetVstoObject(Doc);

                richTextControls = new System.Collections.Generic.List
                    <Microsoft.Office.Tools.Word.RichTextContentControl>();
                int count = 0;

                foreach (Word.ContentControl nativeControl in Doc.ContentControls)
                {
                    if (nativeControl.Type ==
                        Microsoft.Office.Interop.Word.WdContentControlType.wdContentControlRichText)
                    {
                        count++;
                        Microsoft.Office.Tools.Word.RichTextContentControl tempControl =
                            extendedDocument.Controls.AddRichTextContentControl(nativeControl,
                            "VSTORichTextControl" + count.ToString());
                        richTextControls.Add(tempControl);
                    }
                }
            }
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--add-content-controls-to-word-documents_6.vb
    Private richTextControls As New System.Collections.Generic.List _
        (Of Microsoft.Office.Tools.Word.RichTextContentControl)

    Private Sub Application_DocumentOpen(ByVal Doc As Microsoft.Office.Interop.Word.Document) _
        Handles Application.DocumentOpen

        If Doc.ContentControls.Count > 0 Then

            Dim extendedDocument As Document = Globals.Factory.GetVstoObject(Doc)

            Dim count As Integer = 0
            For Each nativeControl As Word.ContentControl In Doc.ContentControls
                If nativeControl.Type = Word.WdContentControlType.wdContentControlRichText Then
                    count += 1
                    Dim tempControl As Microsoft.Office.Tools.Word.RichTextContentControl = _
                        extendedDocument.Controls.AddRichTextContentControl(nativeControl, _
                        "VSTORichTextContentControl" + count.ToString())
                    richTextControls.Add(tempControl)
                End If
            Next nativeControl
        End If
    End Sub